2-Draft-Accepted/RFCxxxx-DSC-for-PowerShell7.md

```diff
@@ -0,0 +1,117 @@
+This enabled executing the Get, Set, and Test methods on DSC resources bypassing the LCM (Local Configuration Manager) on Windows and removing the
+need for OMI ([Open Management Infrastructure](https://github.com/Microsoft/omi)) on Linux.
+This was the start to enable cross platform DSC resources to be authored in PowerShell script (with some limitations).
+
+This feature was shipped as an [experimental feature](https://docs.microsoft.com/powershell/scripting/learn/experimental-features?view=powershell-7) in PowerShell 7.0.
+
+### Bring your own agent
+
+A significant change from existing DSC for Windows PowerShell is there will be no default agent (LCM) shipped with PowerShell 7.
+
+Instead, we want to make it easier to integrate with existing configuration agents like [Ansible](https://docs.ansible.com/ansible/latest/user_guide/windows_dsc.html),
+[Chef](https://docs.chef.io/resources/dsc_resource/), [Puppet](https://forge.puppet.com/puppetlabs/dsc),
+[Azure Policy Guest Configuration](https://docs.microsoft.com/azure/governance/policy/concepts/guest-configuration), and other popular configuration solutions.
+
+Custom agents can simply use the existing PowerShell API (or call out to `pwsh`) to directly invoke DSC resources.
+
+You can even simply write a PowerShell script as your agent that runs off TaskScheduler or cron, for example.
+
+Along with the lack of a default agent, there is no pre-defined push or pull model for deploying DSC.
+
+### Cross platform first
+
+DSC has supported both Windows and Linux, however, the [Linux support](https://docs.microsoft.com/powershell/scripting/dsc/getting-started/lnxGettingStarted?view=powershell-7)
+was dependent on [OMI](https://github.com/Microsoft/omi) and did not support authoring Linux DSC resources in PowerShell script making development
+inconsistent and much more difficult.
+
+DSC for PowerShell 7 will be cross platform having consistent capabilities and user experience on all platforms.
+
+### Moving away from WMI/OMI/MOF
+
+DSC on Windows was originally designed relying on [WMI](https://docs.microsoft.com/windows/win32/wmisdk/wmi-start-page) as a host on Windows,
+OMI as a host on Linux, and [MOF](https://docs.microsoft.com/windows/win32/wmisdk/managed-object-format--mof-) for configuratio and schema.
+
+MOF as a format is not something most developers are familiar with and not much tooling support.
+However, developers of DSC resources written in PowerShell script still needed to learn MOF and also keep their MOF schema
+in sync with their script implementation.
+
+Today, when you execute a [DSC configuration script](https://docs.microsoft.com/powershell/scripting/dsc/configurations/configurations?view=powershell-7),
+the output is a MOF configuration file.
+In general, users do not need to read this MOF file as it's intended to be used by machines.
+However, MOF parsing creates a dependency on WMI/OMI.
+Instead, we will move towards a JSON configuration file format.
```

Comment:
The serialized JSON file format is not the most critical thing IMO.
Having access to the in-memory graph of resource, and if possible on the fly during compilation, can enable some nice scenarios around config data (i.e. JIT credential fetching from vault).

The serialized version of this is great for a common data serialization structure, but we've already "lost" some info (metadata about config data).

Reply:
Additional tooling to understand the dependency graph will have to come later.  We have a partner already that wants the JSON format as they are building their own LCM.  At this point, I think the scope is to define the JSON format and leave it to developers to handle the rest.

Reply:
Whose responsibility will it be to make sure that PSCredential objects don't end-up in plain text in the compiled JSON file if we are to bring our own agent? How do we foresee credential encryption work here?